test/acquisition/multi_objective/test_hypervolume_knowledge_gradient.py

```diff
@@ -304,26 +304,6 @@ def test_evaluate_q_hvkg(self):
             self.assertTrue(
                 torch.equal(qHVKG.extract_candidates(X), X[..., : -n_f * num_pareto, :])
             )
```

Comment:
For cost-aware optimization, could you add a unit test demonstrating that this chooses the right candidate in cases where the logic has changed (negative deltas)?

And for HVKG, is there some way of demonstrating that it's still giving the "right" answer with the clamping removed? It might be good to keep this test in place but just change the assert at the end. I'm not very familiar with HVKG, so this may not turn out to be a good suggestion.

Reply:
Hi @esantorella 

Thanks for the feedback!
I'll add a unit test for demonstrating that it chooses the right candidate tomorrow.

I'm unfortunately not familiar with HVKG either... Any thoughts on this? @sdaulton @Balandat

Reply:
One way would be to run the MFKG tutorial: https://github.com/pytorch/botorch/blob/main/tutorials/Multi_objective_multi_fidelity_BO.ipynb - not sure if this does run into issues with negative improvements, but it's an easy first-order test (you can make a local edit in the code to check whether that new functionality is triggered)

Reply:
Hi @esantorella and @Balandat 

I've added tests that demonstrate the correct functionality in the ```test_cost_aware.py``` file, as I think it makes more sense to have it there. (The alternative being to add tests for it in the tests classes associated with MFKG and HVKG)

I've also run the MFKG tutorial you mentioned @Balandat. Below are some of the printouts for the (negative-only) AF values before and after scaling by the cost:

```
-0.07815102929370721 -> -2.9963066532177565
-0.07778381446031779 -> -7.4577009200049895
-0.08422953299033326 -> -5.449204194716502
...
-0.10146312396850868 -> -0.10146312396850868
-0.10146311575673453 -> -0.10146311575673453
-0.10146311575673453 -> -0.10146311575673453
```

The change also seemed to improve the HVKG results from a regret in the range [-1.55, -1.6] to [-1.6, -1.65]:

Before:
![image](https://github.com/pytorch/botorch/assets/90089300/1f1517b7-9c6c-4a7e-bc63-cf20f88b9c27)

After:
![image](https://github.com/pytorch/botorch/assets/90089300/3ebf1885-3532-4904-b822-31f4ca987d4a)

Should I rerun and add the tutorials that use the ```InverseCostWeightedUtility``` to this PR as well?

Thanks!

Reply:
Oh yeah, it would be great to re-run those tutorials if the results change. Good catch, thanks!